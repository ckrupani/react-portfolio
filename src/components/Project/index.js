import React from "react";

function Project({ projectInfo, imageId }) {
    return (
        <div className="project-wrap">
            <img
                src={require(`../../assets/small/${imageId}.jpg`)}
                alt={projectInfo.name}
                className="img-thumbnail mx-1"
            />
            <div className="project-description">
                <h3>
                    <a
                        href={projectInfo.appLink}
                        target="_blank"
                        rel="noreferrer"
                    >
                        {projectInfo.name}
                    </a>
                </h3>
                <p>{projectInfo.description}</p>
                <a href={projectInfo.repoLink} target="_blank" rel="noreferrer">
                    Github
                </a>
            </div>
        </div>
    );
}

export default Project;
