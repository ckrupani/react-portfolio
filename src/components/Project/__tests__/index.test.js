import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Project from "../";

afterEach(cleanup);

describe("Project is rendering", () => {
    const projectInfo = {
        name: "Food Store",
        description: "It is a webportal to buy food.",
        appLink: "",
        repoLink: "",
    };

    it("renders", () => {
        render(<Project projectInfo={projectInfo} imageId={0} />);
    });

    it("matches the snapshot", () => {
        const { asFragment } = render(
            <Project projectInfo={projectInfo} imageId={0} />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
