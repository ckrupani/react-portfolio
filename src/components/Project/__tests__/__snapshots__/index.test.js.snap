// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project is rendering matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="project-wrap"
  >
    <img
      alt="Food Store"
      class="img-thumbnail mx-1"
      src="0.jpg"
    />
    <div
      class="project-description"
    >
      <h3>
        <a
          href=""
          rel="noreferrer"
          target="_blank"
        >
          Food Store
        </a>
      </h3>
      <p>
        It is a webportal to buy food.
      </p>
      <a
        href=""
        rel="noreferrer"
        target="_blank"
      >
        Github
      </a>
    </div>
  </div>
</DocumentFragment>
`;
